<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>913</width>
    <height>643</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,2">
    <item>
     <layout class="QGridLayout" name="gridLayout_3" rowstretch="0" columnstretch="0,0">
      <item row="0" column="1">
       <widget class="QWidget" name="widget" native="true">
        <layout class="QVBoxLayout" name="verticalLayout_5" stretch="1,1,1,0,0">
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>Parameters</string>
           </property>
           <layout class="QFormLayout" name="formLayout_6">
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Board Size</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="spinBoxN"/>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Efficient Mode</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QCheckBox" name="checkBoxEff">
              <property name="text">
               <string>On</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0" colspan="2">
             <widget class="QPushButton" name="pushButtonGenerate">
              <property name="text">
               <string>Generate Grid</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_3">
           <property name="title">
            <string>File</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_4" columnstretch="0,0,0">
            <item row="0" column="2">
             <widget class="QPushButton" name="pushButtonBrowse">
              <property name="text">
               <string>Browse</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0" colspan="3">
             <widget class="QPushButton" name="pushButtonSave">
              <property name="text">
               <string>Save</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0" colspan="3">
             <widget class="QPushButton" name="pushButtonImport">
              <property name="text">
               <string>Import</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Path</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="lineEditPath"/>
            </item>
            <item row="1" column="1" colspan="2">
             <widget class="QComboBox" name="comboBoxMode">
              <item>
               <property name="text">
                <string>Text (.txt)</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Image (.png .jpg .bmp)</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Mode</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_2">
           <property name="title">
            <string>Status</string>
           </property>
           <layout class="QFormLayout" name="formLayout_7">
            <item row="0" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Iteration</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="labelIteration">
              <property name="autoFillBackground">
               <bool>false</bool>
              </property>
              <property name="styleSheet">
               <string notr="true"/>
              </property>
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Time</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLabel" name="labelTime">
              <property name="text">
               <string>-</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>State</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QLabel" name="labelState">
              <property name="text">
               <string>Idle</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButtonSolve">
           <property name="text">
            <string>Solve</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButtonStop">
           <property name="text">
            <string>Stop</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="boardWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <property name="spacing">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="inputPage">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QTableWidget" name="tableWidgetColor"/>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonChangeColor">
          <property name="text">
           <string>Change Color</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonBuildBoard">
          <property name="text">
           <string>Build Board</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="imagePage">
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="ImageWidget" name="imageWidget" native="true"/>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonBuildBoardInput">
          <property name="text">
           <string>Build Board</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>913</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ImageWidget</class>
   <extends>QWidget</extends>
   <header>helper/ImageWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
